mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/ComponentParameterization.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.component;
+
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.extension.internal.component.ComponentParameterizationBuilder;
+
+/**
+ * Represents a specific configuration for a concrete {@link ParameterizedModel}.
+ * <p>
+ * Instances are validated during creation, so no instances must contain invalid parameters according to the {@link #getModel()
+ * model}.
+ *
+ * @param <M> the actual {@link ParameterizedModel} sub-type of the component .
+ * 
+ * @since 1.5
+ */
+public interface ComponentParameterization<M extends ParameterizedModel> {
```

Comment:
this should also apply for configs and connectino providers, which are not components

Reply:
what do you mean they are not components? they are not processors, but components yes.

Reply:
per extensions-api terminology, component is something that handles events (operations, sources , routers, etc). Configs and CP don't fit in that category. But we can adjust it in another PR after discussion.